workflow/executor/pns/pns.go

```diff
@@ -368,18 +369,25 @@ func (p *PNSExecutor) updateCtrIDMap() {
 	}
 }
 
-func (p *PNSExecutor) GetContainerStatus(containerID string) (*corev1.Pod, *corev1.ContainerStatus, error) {
-	pod, err := p.clientset.CoreV1().Pods(p.namespace).Get(p.podName, metav1.GetOptions{})
-	if err != nil {
-		return nil, nil, fmt.Errorf("could not get pod: %s", err)
-	}
-	for _, containerStatus := range pod.Status.ContainerStatuses {
-		if execcommon.GetContainerID(&containerStatus) != containerID {
-			continue
+func (p *PNSExecutor) GetTerminatedContainerStatus(containerID string) (*corev1.Pod, *corev1.ContainerStatus, error) {
+	var pod *corev1.Pod
+	var containerStatus *corev1.ContainerStatus
+	err := wait.Poll(1*time.Second, 3*time.Second, func() (bool, error) {
```

Comment:
is this try 3 times over 3 seconds?

Reply:
Correct. According to my manual testing, this should succeed on the first try around 95% of the time. It should almost always succeed by the second try.